airbyte-cdk/python/unit_tests/sources/declarative/migrations/test_legacy_to_per_partition_migration.py

```diff
@@ -241,3 +242,33 @@ def _migrator_with_multiple_parent_streams():
     config = {}
     parameters = {}
     return LegacyToPerPartitionStateMigration(partition_router, cursor, config, parameters)
+
+
+def _migrator_with_custom_partition_router():
```

Comment:
is this used? we should probably add 2 unit tests
1. with a custom partition router that has parent_stream_configs
2. with a custom partition router that does not have a parent_stream_configs

Reply:
Added